src/java.base/share/classes/jdk/internal/foreign/SegmentFactories.java

```diff
@@ -203,7 +203,7 @@ private static long allocateNativeInternal(long byteSize, long byteAlignment, Me
         long allocationSize;
         long allocationBase;
         long result;
-        if (byteAlignment > MAX_MALLOC_ALIGN) {
+        if (byteAlignment > MAX_MALLOC_ALIGN || alignedSize % byteAlignment != 0) {
             allocationSize = alignedSize + byteAlignment - MAX_MALLOC_ALIGN;
```

Comment:
The calculation of `allocationSize` looks no longer correct now that `byteAlignment` might be smaller than `MAX_MALLOC_ALIGN`, and it can result in a negative size.